controllers/productController.test.js

```diff
@@ -0,0 +1,545 @@
+                quantity: 10,
+                shipping: true
+        };
+        req.files = {
+                photo: {
+                path: 'client/public/images/test-pdt-img-1.jpg', 
+                size: 1024 
+                }
+            };
+        req.params = {pid: 1}
+
+        productModel.findByIdAndUpdate =jest.fn().mockReturnValue(null);
+
+        await updateProductController(req, res);
+        expect(res.status).toHaveBeenCalledWith(500);
+        expect(res.send).toHaveBeenCalledWith({
+            error: 'Description is Required'
+        });
+    });
+
+    test('should return 500 with correct error message if no price in req body', async () => {
+        req.fields = {
+                name: 'Test Name',
+                description: 'Test Description',
+                category: 'Test Category',
+                quantity: 10,
+                shipping: true
+        };
+        req.files = {
+                photo: {
+                path: 'client/public/images/test-pdt-img-1.jpg', 
+                size: 1024 
+                }
+            };
+
+        req.params = {pid: 1}
+
+        await updateProductController(req, res);
+        expect(res.status).toHaveBeenCalledWith(500);
+        expect(res.send).toHaveBeenCalledWith({
+            error: 'Price is Required'
+        });
+    });
+
+    test('should return 500 with correct error message if no quantity in req body', async () => {
+        req.fields = {
+            name: 'Test Name',
+            description: 'Test Description',
+            category: 'Test Category',
+            price: 100,
+            shipping: true
+        };
+        req.files = {
+                photo: {
+                path: 'client/public/images/test-pdt-img-1.jpg', 
+                size: 1024 
+                }
+        };
+
+        req.params = {pid: 1}
+
+        await updateProductController(req, res);
+        expect(res.status).toHaveBeenCalledWith(500);
+        expect(res.send).toHaveBeenCalledWith({
+            error: 'Quantity is Required'
+        });
+    });
+
+    test('should return 500 with correct error message if no category in req body', async () => {
+        req.fields = {
+            name: 'Test Name',
+            description: 'Test Description',
+            quantity: 10,
+            price: 100,
+            shipping: true
+        };
+        req.files = {
+                photo: {
+                path: 'client/public/images/test-pdt-img-1.jpg', 
+                size: 1024 
+                }
+        };
+
+        req.params = {pid: 1}
+
+        await updateProductController(req, res);
+        expect(res.status).toHaveBeenCalledWith(500);
+        expect(res.send).toHaveBeenCalledWith({
+            error: 'Category is Required'
+        });
+    });
+
+    test('should return 500 with correct error message if no photo in req files', async () => {
+        // supposed to fail due to error in original implementation
```

Comment:
What is the error in original implementation? Is it should be `!photo` instead of `photo`?

Reply:
I believe there is an inconsistency in the check and the corresponding error message - pertaining to whether photo is meant to be optional or not. My current understanding is that the photo is "required" (i.e. correct error message but implementation mistake) - however the same assumption does not seem to cascade downwards in implementation - will adopt one of these as bugs upon closer examination with rational documented

Reply:
Alright! My understanding is same as yours, where a photo should be required. 👍